website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,206 @@
 :::
 
+<VersionBlock lastVersion="1.7">
+
+Warnings that should be treated as errors can be specified through `include` and/or `exclude` parameters. Warning names can be found in:
+- [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`) 
+- Downloading the JSON output logs from a run and searching for the warning.
+
+The `include` parameter can be set to "all" or "*" to treat all warnings as exceptions, or to a list of specific warning names to treat as exceptions. When include is set to "all" or "*", the optional exclude parameter can be set to exclude specific warnings from being treated as exceptions.
+
+  :::caution Proceed with caution in production environments
+
+  Using `warn_error_options: include: "all"` will treat _all_ current and future warnings as errors.
+
+  This means that if a new warning is introduced in a future version of dbt Core, your production job may start failing unexpectedly. We recommend proceeding with caution when doing this in production environments, and explicitly listing only the warnings you want to treat as errors in production.
+
+  :::
+
+</VersionBlock>
+
 <VersionBlock firstVersion="1.8">
 
-- Warnings that should be treated as errors can be specified through `error` and/or `warn` parameters. Warning names can be found in [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`).
+- Warnings that should be treated as errors can be specified through `error` and/or `warn` parameters. Warning names can be found in:
+- [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`).
+- Downloading the JSON output logs from a run and searching for the warning.
 
-- The `error` parameter can be set to `"all"` or `"*"` to treat all warnings as exceptions, or to a list of specific warning names to treat as exceptions. When `error` is set to `"all"` or `"*"`, the optional `warn` parameter can be set to exclude specific warnings from being treated as exceptions.
+- The `error` parameter can be set to `"all"` or `"*"` to treat all warnings as exceptions (errors), or to a list of specific warning names to treat as exceptions. When `error` is set to `"all"` or `"*"`, the optional `warn` parameter can be set to exclude specific warnings from being treated as exceptions.
```

Comment:
maybe worth calling out that promoting all warnings to errors is the same as using `--warn-error`

Reply:
but using `--warn-error-options` allows you to then exclude specific warnings with `warn` parameter